dll/win32/browseui/aclistisf.cpp

```diff
@@ -339,12 +345,16 @@ STDMETHODIMP CACListISF::Expand(LPCOLESTR pszExpand)
     {
         if (PathIsRelativeW(pszExpand) &&
             SHGetPathFromIDListW(m_pidlCurDir, szPath1) &&
-            PathCombineW(szPath2, szPath1, pszExpand))
+            PathCombineW(szPath2, szPath1, pszExpand) &&
+            PathFileExistsW(szPath2))
         {
             pszExpand = szPath2;
         }
-        GetFullPathNameW(pszExpand, _countof(szPath1), szPath1, NULL);
-        pszExpand = szPath1;
+        else if (PathFileExistsW(pszExpand))
```

Comment:
So that I understand:
you now don't call GetFullPathNameW on the pszExpand that could have been set in the previous if-block, so that e.g. specific shell folder names like `shell:windows\...` aren't attempted to be expanded?

Reply:
`shell:windows\...` is not a file.
Passing an invalid filename to `GetFullPathNameW` will generate an invalid pathname.

Reply:
> `shell:windows\...` is not a valid file pathname

I guess it does not apply here but for future reference,  `shell:windows\foo` can be a valid shell path.

Reply:
Yes, `shell:windows\foo` is a shell (shell32 and the like)-only construct; this is not obviously recognized to win32 kernel32 `GetFullPathNameW` function (and the NT `RtlGetFullPathName_U` one), so _**not**_ passing that shell-specific thing to these functions is the best course of action.